app/assets/javascripts/editor.js

```diff
@@ -1,13 +1,21 @@
-// jQuery (document).ready function:
-
-$E = {
-  initialize: function() {
-    // call setState with no parameters, aka. default parameters.
-    // default parameters point toward either:
-    //   1. the comment form at the bottom of multi-comment wikis/questions/research notes
-    //   2. the only editor form on /wiki/new and /wiki/edit
-    $E.setState();
-    
+class Editor {
+  // default parameters reference the IDs of:
+  //   1. main comment form in multi-comment wikis, questions, & research notes.
+  //   2. the only editor form on /wiki/new and /wiki/edit
+  constructor(textarea = "text-input", preview = "comment-preview-main", title = "title") {
```

Comment:
These default params are just a stepping stone for now. 

Since the element IDs are all standardized now, it's easier to do a rewrite so that constructor takes just **one** parameter. For example: 

`$E = new Editor('reply-1234');` means that the editor object has access to both `#comment-preview-**reply-1234**` and `#text-input-**reply-1234**`. 

Much simpler.

Reply:
Nice !!